jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -111,11 +113,49 @@ public boolean handle(Request request, Response response, Callback callback) thr
             return false;
         }
         Handler handler = matchedResource.getResource();
+        PathSpec pathSpec = matchedResource.getPathSpec();
         if (LOG.isDebugEnabled())
             LOG.debug("Matched {} to {} -> {}", pathInContext, matchedResource.getPathSpec(), handler);
-        boolean handled = handler.handle(request, response, callback);
+
+        PathSpecRequest pathSpecRequest = new PathSpecRequest(request, pathSpec);
+        boolean handled = handler.handle(pathSpecRequest, response, callback);
```

Comment:
I now definitely think we should make this optional.   It may be that a PathMappingsHandler is being used to map to "contexts"... in which case this request/context wrapping is good.    But it might also be used within a context, so that makes no sense.

This should either be optional behaviour on this handler or moved to a subclass `PathMappingsHandler.ContextMapping`